### DISCUSSION TITLE
How To Use Latexindent On Linux

### DISCUSSION THREAD

**Discussion by niekdomi** [texstudio-org/texstudio#3905] [Q&A] [answered]
Hey, how can I use `latexindent` on linux.

I added a usercommand under `Options -> Configure Texstudio -> Build` 

with the following content: `latexindent -m %.tex`

But I have following issues:
1. It runs the main.tex not the current opened file
2. The command runs in the terminal and doesn't format the file in the editor

**Comment by sunderme** [comment]
see https://texstudio-org.github.io/configuration.html#command-syntax-in-detail
I would suggest to define a user command.
commands in general can only transform file content, hence after latexident, you need to reload the file (though txs usually detects changes on disk automatically and asks if you want to reload)

**Comment by niekdomi** [comment]
The command `latexindent -m "?c:ame)"` successfully runs `latexindent` for the current file but doesn't apply the formatting directly to the file in the editor. Instead, it outputs the formatted content to the message window.

This means the formatting changes aren't saved back to the file itself.

Is there a way to fix this?

**Comment by octaeder** [comment]
From the manual:

![grafik](https://github.com/user-attachments/assets/f154491e-1fb1-489c-9b18-abba267972ac)
![grafik](https://github.com/user-attachments/assets/91fa34b6-55a7-4322-928e-c510a910fe51)

**Comment by niekdomi** [comment, marked as answer]
Alright, thank you. didn't know an additional flag was needed, because in neovim it works without `-w`

Here's the final command: `latexindent -w -m "?c:ame)"`
